modules/dxp/apps/scim/scim-configuration-web-test/src/testIntegration/java/com/liferay/scim/configuration/web/internal/portlet/action/test/SaveScimConfigurationMVCActionCommandTest.java

```diff
@@ -0,0 +1,143 @@
+import com.liferay.portal.kernel.test.rule.DeleteAfterTestRun;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.kernel.test.util.UserTestUtil;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.Constants;
+import com.liferay.portal.kernel.util.HashMapDictionaryBuilder;
+import com.liferay.portal.kernel.util.WebKeys;
+import com.liferay.portal.test.rule.FeatureFlags;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.scim.rest.util.ScimClientUtil;
+
+import java.util.List;
+
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Christian Moura
+ */
+@FeatureFlags("LPS-96845")
+@RunWith(Arquillian.class)
+public class SaveScimConfigurationMVCActionCommandTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	@AfterClass
+	public static void tearDownClass() throws Exception {
+		ConfigurationTestUtil.deleteConfiguration(_pid);
+	}
+
+	@Test
+	public void testOAuth2TokenIsIssuedToCurrentUser() throws Exception {
+		_user = UserTestUtil.addCompanyAdminUser(
+			CompanyLocalServiceUtil.getCompany(TestPropsValues.getCompanyId()));
+
+		_pid = ConfigurationTestUtil.createFactoryConfiguration(
+			"com.liferay.scim.rest.internal.configuration." +
+				"ScimClientOAuth2ApplicationConfiguration",
+			HashMapDictionaryBuilder.<String, Object>put(
+				"companyId", TestPropsValues.getCompanyId()
+			).put(
+				"matcherField", "email"
+			).put(
+				"oAuth2ApplicationName", "TEST Scim client"
+			).put(
+				"userId", _user.getUserId()
+			).build());
+
+		String scimClientId = ScimClientUtil.generateScimClientId(
+			"TEST Scim client");
+
+		OAuth2Application oAuth2Application =
+			_oAuth2ApplicationLocalService.getOAuth2Application(
+				TestPropsValues.getCompanyId(), scimClientId);
+
+		Assert.assertNotNull(oAuth2Application);
+
+		MockLiferayPortletActionRequest mockLiferayPortletActionRequest =
+			new MockLiferayPortletActionRequest();
+
+		mockLiferayPortletActionRequest.addParameter(Constants.CMD, "generate");
+
+		mockLiferayPortletActionRequest.addParameter(
+			"oAuth2ApplicationName", "TEST Scim client");
+
+		ThemeDisplay themeDisplay = new ThemeDisplay();
+
+		themeDisplay.setCompany(
+			CompanyLocalServiceUtil.fetchCompany(
+				TestPropsValues.getCompanyId()));
+		themeDisplay.setUser(_user);
+
+		themeDisplay.setPermissionChecker(
+			PermissionCheckerFactoryUtil.create(_user));
+
+		mockLiferayPortletActionRequest.setAttribute(
+			WebKeys.THEME_DISPLAY, themeDisplay);
+
+		_mvcActionCommand.processAction(
+			mockLiferayPortletActionRequest,
+			new MockLiferayPortletActionResponse());
+
+		List<OAuth2Authorization> authorizations =
```

Comment:
Rename to `oAuth2Authorizations`, so the model and the list type are named similarly.